h2o-webserver-iface/src/main/java/water/webserver/iface/H2OHttpView.java

```diff
@@ -0,0 +1,38 @@
+package water.webserver.iface;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+
+/**
+ * Exposes part of H2O functionality for the purposes of HTTP server adapter.
+ * Contains also logic for handling authentication and other functionality, so that it can be easily shared among various server implementations.
+ */
+public interface H2OHttpView {
+
+  /**
+   * @return configuration related to HTTP server
+   */
+  H2OHttpConfig getConfig();
+
+  /**
+   * @return map of servlets with their context paths
+   */
+  LinkedHashMap<String, Class<? extends HttpServlet>> getServlets();
```

Comment:
Do we really need to return `LinkedHashMap` ? Please specify if the returned Map is immutable or not. 

Currently, it only returns pointer to the original `LinkedHashMap`, which is fully mutable. I wouldn't expose it. Let's make it immutable or return a copy at least.

Reply:
I didn't use `Map` because I wanted to express at API level that the order matters - at least, I cannot safely say it does not.
*Immutable* is a good idea, but the only way to make it here is, exposing a `List` instead (even `Iterable` would be enough) of registry entries which would be a new class; this is probably also much cleaner. Will do that, thanks for pointing out.